GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Submodules.cs

```diff
@@ -0,0 +1,431 @@
+
+                string GetSubmoduleSuperPath(string submodulePath)
+                {
+                    var superPath = modulePaths.Find(path => submodulePath != path && submodulePath.Contains(path));
+                    Trace.Assert(superPath != null);
+                    return superPath;
+                }
+            }
+
+            private string GetNodeRelativePath(GitModule topModule, SubmoduleNode node)
+            {
+                return node.SuperPath.SubstringAfter(topModule.WorkingDir).ToPosixPath() + node.LocalPath;
+            }
+
+            private void AddNodesToTree(List<SubmoduleNode> nodes)
+            {
+                if (!UseFolderTree)
+                {
+                    Nodes.AddNodes(nodes);
+                    return;
+                }
+                else
+                {
+                    // Create tree of SubmoduleFolderNode for each path directory and add input SubmoduleNodes as leaves.
+
+                    // Example of (SuperPath + LocalPath).ToPosixPath() for all nodes:
+                    //
+                    // C:/code/gitextensions2/Externals/conemu-inside
+                    // C:/code/gitextensions2/Externals/Git.hub
+                    // C:/code/gitextensions2/Externals/ICSharpCode.TextEditor
+                    // C:/code/gitextensions2/Externals/ICSharpCode.TextEditor/gitextensions
+                    // C:/code/gitextensions2/Externals/ICSharpCode.TextEditor/gitextensions/Externals/conemu-inside
+                    // C:/code/gitextensions2/Externals/ICSharpCode.TextEditor/gitextensions/Externals/Git.hub
+                    // C:/code/gitextensions2/Externals/ICSharpCode.TextEditor/gitextensions/Externals/ICSharpCode.TextEditor
+                    // C:/code/gitextensions2/Externals/ICSharpCode.TextEditor/gitextensions/Externals/NBug
+                    // C:/code/gitextensions2/Externals/ICSharpCode.TextEditor/gitextensions/GitExtensionsDoc
+                    // C:/code/gitextensions2/Externals/NBug
+                    // C:/code/gitextensions2/GitExtensionsDoc
+                    //
+                    // What we want to do is first remove the topModule portion, "C:/code/gitextensions2/", and
+                    // then build our tree by breaking up each path into parts, separated by '/'.
+                    //
+                    // Note that when we break up the paths, some parts are just directories, the others are submodule nodes:
+                    //
+                    // Externals / ICSharpCode.TextEditor / gitextensions / Externals / Git.hub
+                    //  folder          submodule             submodule      folder     submodule
+                    //
+                    // Input 'nodes' is an array of SubmoduleNodes for all the submodules; now we need to create SubmoduleFolderNodes
+                    // and insert everything into a tree.
+
+                    GitModule topModule = Module.GetTopModule();
```

Comment:
I've seen this in few places, probably better to use `var`

Reply:
I only found the one you pointed out, and replaced with using var.